op-node/rollup/derive/frame_queue.go

```diff
@@ -20,12 +22,14 @@ type FrameQueue struct {
 	log    log.Logger
 	frames []Frame
 	prev   NextDataProvider
+	cfg    *rollup.Config
 }
 
-func NewFrameQueue(log log.Logger, prev NextDataProvider) *FrameQueue {
+func NewFrameQueue(log log.Logger, cfg *rollup.Config, prev NextDataProvider) *FrameQueue {
```

Comment:
nit: I wonder if we should have a smaller interface to use here rather than requiring the specific `rollup.Config` type when we only need the `IsHolocene` method.  I've often found writing tests its annoying to have to come up with a whole config.

Might not be worth it for this case since go interfaces tend to be declare where they're used - supporting a more general interface on `rollup.Config` with just the hard fork check methods seems appealing.

Reply:
Yep that's true, it would be nice in tests to be able to only mock the parts of the config, or `ChainSpec`, that are really consumed. But I'd prefer to leave this as future cleanup. Here, I just went with the existing pattern - all other stages also just receive a `*rollup.Config` instead of an interface.
https://github.com/ethereum-optimism/optimism/blob/32a3637ef893e656fa4526880a4c3fa428115ccc/op-node/rollup/derive/pipeline.go#L82-L91